### PR TITLE
Refactored db.py to use query_db.

### PR DESCRIPTION
**Pull request by jteague9**

### CODE REVIEW

File: db.py [hunk 1]
```diff
@@ -34,45 +36,37 @@ def init_ladder_db():
...
+
+def query_db(query, arguments, single=False):
     db = psycopg2.connect(dbname=dbname, user=user, password=password, host=host, port=port)
     c = db.cursor()
-    c.execute('''SELECT * FROM users WHERE bnet_id ILIKE %s OR username ILIKE %s LIMIT %s''',
-        [username + '%', username + '%', str(limit)])
-    players = c.fetchall()
+    c.execute(query, arguments)
+    if single:
+        result = c.fetchone()[0]
```
Comment (tso) [inline db.py]: this seems weird to me

File: db.py [hunk 2]
```diff
@@ -34,45 +36,37 @@ def init_ladder_db():
...
+    else:
+        result = c.fetchall()
     db.close()
-    return players
+    return result
+
+
+def search_player_by_name(username, limit=25):
+    return query_db('''SELECT * FROM users WHERE bnet_id ILIKE %s OR username ILIKE %s LIMIT %s''',
+                    [username + '%', username + '%', str(limit)])
```
Comment (tso) [inline db.py]: I don't think we need to do `str(limit)`

File: db.py [hunk 3]
```diff
@@ -34,45 +36,37 @@ def init_ladder_db():
...
 def get_count_in_region(region):
-    db = psycopg2.connect(dbname=dbname, user=user, password=password, host=host, port=port)
-    c = db.cursor()
-    c.execute('''SELECT count(*) FROM users WHERE REGION=%s''',
-        (region,))
-    count = c.fetchone()
-    db.close()
-    return count[0]
+    return query_db('''SELECT count(*) FROM users WHERE REGION=%s''',
+                    (region,), single=True)
```
Comment (tso) [inline db.py]: prefer `[region]` over `(region,)` personally

File: db.py (hunk 2 above)
Comment (jteague9) [reply db.py]: Fixed

File: db.py (hunk 1 above)
Comment (jteague9) [reply db.py]: Updated

File: db.py (hunk 3 above)
Comment (jteague9) [reply db.py]: Fixed

**Review by tso — APPROVED** [review]
lgtm

**Review by EugeneLu12 — APPROVED** [review]

**Pull request merged**